ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -286,22 +283,22 @@ addBlockSync cdb@CDB {..} BlockToAdd { blockToAdd = b, .. } = do
     -- ### Ignore
     newTip <- if
       | olderThanK hdr isEBB immBlockNo -> do
-        traceWith addBlockTracer $ IgnoreBlockOlderThanK (blockRealPoint b)
-        deliverWrittenToDisk False
+        lift $ traceWith addBlockTracer $ IgnoreBlockOlderThanK (blockRealPoint b)
+        lift $ deliverWrittenToDisk False
         chainSelectionForFutureBlocks cdb BlockCache.empty
 
       | isMember (blockHash b) -> do
-        traceWith addBlockTracer $ IgnoreBlockAlreadyInVolatileDB (blockRealPoint b)
-        deliverWrittenToDisk True
+        lift $ traceWith addBlockTracer $ IgnoreBlockAlreadyInVolatileDB (blockRealPoint b)
+        lift $ deliverWrittenToDisk True
         chainSelectionForFutureBlocks cdb BlockCache.empty
 
       | Just (InvalidBlockInfo reason _) <- Map.lookup (blockHash b) invalid -> do
-        traceWith addBlockTracer $ IgnoreInvalidBlock (blockRealPoint b) reason
-        deliverWrittenToDisk False
+        lift $ traceWith addBlockTracer $ IgnoreInvalidBlock (blockRealPoint b) reason
+        lift $ deliverWrittenToDisk False
 
         -- We wouldn't know the block is invalid if its prefix was invalid,
         -- hence 'InvalidBlockPunishment.BlockItself'.
-        InvalidBlockPunishment.enact
+        lift $ InvalidBlockPunishment.enact
           blockPunish
           InvalidBlockPunishment.BlockItself
```

Comment:
Could also move the `lift` before the `do` here.

Reply:
Ah indeed

Reply:
Ah no we can't! there is a `chainSelectionForFutureBlocks` below!

Reply:
Ah sorry, I was tricked by exactly these two lines being hidden in the GH UI :laughing: